File: app/src/main/java/com/duy/interpreter/exceptions/grouping/StrayCharacterException.java
package com.duy.interpreter.exceptions.grouping;

import com.duy.interpreter.exceptions.grouping.*;
import com.duy.interpreter.linenumber.LineInfo;

public class StrayCharacterException extends com.duy.interpreter.exceptions.grouping.GroupingException {

    public StrayCharacterException(LineInfo line, char character) {
        super(line, "Stray character in program: <" + character
                + ">, char code " + (int) character);
    }

}

File: app/src/main/java/com/duy/pascal/interperter/parse_exception/grouping/StrayCharacterException.kt
package com.duy.pascal.interperter.parse_exception.grouping

import com.duy.pascal.interperter.linenumber.LineInfo

class StrayCharacterException : GroupingException {
    var charCode: Char

    constructor(line: LineInfo, charCode: Char) : super(line,
            "Stray character in program: $charCode\nChar code ${charCode.toInt()}") {
        this.charCode = charCode
    }
}

